torch/distributed/_tensor/device_mesh.py

```diff
@@ -105,7 +105,7 @@ def __init__(
     ) -> None:
         self.device_type = device_type
         self.mesh = (
-            mesh.detach()
+            mesh
```

Comment:
then doesn't the original `mesh.detach()` remove `mesh` from the graph? Why should we change it to `mesh` instead?

Reply:
I think detach was there because I want to avoid user modifying the mesh tensor. Removed it because of only hashing the mesh tensor, but since we should force equality, mesh + id should be hashed, so I added the detach() call back :)